### PR TITLE
[Limit Orders]: Design QA Changes

### PR DESCRIPTION
**Pull request by crnbarr93**
## What is the purpose of the change:
These changes are for minor design adjustments across the board. The majority of changes were in the denom selection and limit input components. An entire list of the changes can be found in the linear task.

A few tasks are left as bold as they will be part of other PRs in future.

### Linear Task

[FE-810: Design QA](https://linear.app/osmosis/issue/FE-810/design-qa)

## Brief Changelog
- Several design changes to denom selection
- Several design changes to limit input
- Limit input validation changes

### CODE REVIEW

File: packages/web/components/layouts/main.tsx [hunk 1]
```diff
@@ -60,7 +56,6 @@ export const MainLayout = observer(
         </div>
         <NavBar
           className="ml-sidebar md:ml-0"
-          title={selectedMenuItem?.label ?? ""}
```
Comment (fabryscript) [inline packages/web/components/layouts/main.tsx:63]: i think we don't need to remove every title on every page, you can use the `useNavbar` hook inside the selected page to manipulate the current title

File: packages/web/components/navbar/index.tsx [hunk 2]
```diff
@@ -185,7 +184,7 @@ export const NavBar: FunctionComponent<
       <>
         <div
           className={classNames(
-            "fixed z-[60] flex h-navbar w-[calc(100vw_-_14.58rem)] place-content-between items-center bg-osmoverse-900 px-8 shadow-md lg:gap-5 md:h-navbar-mobile md:w-full md:place-content-start md:px-4",
```
Comment (fabryscript) [inline packages/web/components/navbar/index.tsx:188]: you can pass something like `!shadow-none` in the className prop in order to do this, no need to edit the global navbar

File: packages/web/components/navbar/index.tsx [hunk 3]
```diff
@@ -254,9 +253,6 @@ export const NavBar: FunctionComponent<
             </Popover>
           </div>
           <div className="flex shrink-0 grow items-center gap-9 lg:gap-2 md:place-content-between md:gap-1">
-            <h4 className="md:text-h6 md:font-h6">
```
Comment (fabryscript) [inline packages/web/components/navbar/index.tsx:257]: same as above

File: packages/web/components/place-limit-tool/index.tsx [hunk 4]
```diff
@@ -120,7 +140,12 @@ export const PlaceLimitTool: FunctionComponent<PlaceLimitToolProps> = observer(
     };
 
     const buttonText = useMemo(() => {
-      if (swapState.error) {
+      if (
+        swapState.error &&
+        swapState.error !== "errors.zeroAmount" &&
+        swapState.error !== "errors.emptyAmount" &&
+        swapState.error !== "limitOrders.insufficientFunds"
```
Comment (fabryscript) [inline packages/web/components/place-limit-tool/index.tsx:148]: can't you invert this whole chain of "!==" into one expression of "===" with the only leftover condition?

**Review by fabryscript — CHANGES_REQUESTED** [review]

File: packages/web/components/place-limit-tool/index.tsx (hunk 4 above)
Comment (crnbarr93) [reply packages/web/components/place-limit-tool/index.tsx]: Isn't exclusion simpler here? If we add more errors to the `swapState` we would have to update this every time right?

Comment (fabryscript) [reply packages/web/components/place-limit-tool/index.tsx]: I do not get if this boolean chains aims to isolate a specific error or not

Comment (crnbarr93) [reply packages/web/components/place-limit-tool/index.tsx]: We don't want to display error text for certain errors right? Like Zero amount input/Empty input/Insufficient balance

Comment (crnbarr93) [reply packages/web/components/place-limit-tool/index.tsx]:
```typescript
const NON_DISPLAY_ERRORS = [
  "errors.zeroAmount",
  "errors.emptyAmount",
  "limitOrders.insufficientFunds",
];

const buttonText = useMemo(() => {
  if (swapState.error && !NON_DISPLAY_ERRORS.includes(swapState.error)) {
    return t(swapState.error);
  } else {
    return orderDirection === "bid"
      ? t("portfolio.buy")
      : t("limitOrders.sell");
  }
}, [orderDirection, swapState.error, t]);
```

How about something like this?

**Comment by crnbarr93** [comment]
Closing, covered in other PRs post redesign.

**Pull request closed without merging**